WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m

```diff
@@ -137,6 +137,13 @@ - (BOOL)fb_hasAnyVisibleLeafs
 
 - (BOOL)fb_isVisible
 {
+  if ([FBConfiguration shouldLoadSnapshotWithAttributes]) {
```

Comment:
isn't this similar to the `[(NSNumber *)[self fb_attributeValue:FB_XCAXAIsVisibleAttribute] boolValue]` call below?
Also, caching might still be useful

Reply:
The difference is that `[self fb_attributeValue:FB_XCAXAIsVisibleAttribute]` make a RPC to the accessibility layer, and this happens for each XCElementSnapshot instance. But if we call `fb_snapshotWithAttributes` the visibility value is stored in the XCElementSnapshot instance already (in the `additionalAttributes` dictionary) with this single call to the accessibility layer, so caching is also not required here.

Reply:
It would be nice to add a comment there. Although, I'm still not quite sure about the general stability of this strategy (see the comment below regarding FB_XCAXAIsVisibleAttribute)